sros2/sros2/api/_key.py

```diff
@@ -0,0 +1,139 @@
+    public_certs = ['identity_ca.cert.pem', 'permissions_ca.cert.pem']
+    for public_cert in public_certs:
+        dst = os.path.join(key_dir, public_cert)
+        keystore_ca_cert_path = os.path.join(
+            _keystore.get_keystore_public_dir(keystore_path), public_cert)
+        relativepath = os.path.relpath(keystore_ca_cert_path, key_dir)
+        _utilities.create_symlink(src=relativepath, dst=dst)
+
+    # symlink the governance file in there
+    keystore_governance_path = os.path.join(
+        _keystore.get_keystore_context_dir(keystore_path), 'governance.p7s')
+    dest_governance_path = os.path.join(key_dir, 'governance.p7s')
+    relativepath = os.path.relpath(keystore_governance_path, key_dir)
+    _utilities.create_symlink(src=relativepath, dst=dest_governance_path)
+
+    keystore_identity_ca_cert_path = os.path.join(
+        _keystore.get_keystore_public_dir(keystore_path), 'identity_ca.cert.pem')
+    keystore_identity_ca_key_path = os.path.join(
+        _keystore.get_keystore_private_dir(keystore_path), 'identity_ca.key.pem')
+
+    cert_path = os.path.join(key_dir, 'cert.pem')
+    key_path = os.path.join(key_dir, 'key.pem')
+    if not os.path.isfile(cert_path) or not os.path.isfile(key_path):
+        print('creating cert and key')
+        _create_key_and_cert(
+            keystore_identity_ca_cert_path,
+            keystore_identity_ca_key_path,
+            identity,
+            cert_path,
+            key_path
+        )
+    else:
+        print('found cert and key; not creating new ones!')
+
+    # create a wildcard permissions file for this node which can be overridden
+    # later using a policy if desired
+    policy_file_path = get_policy_default('policy.xml')
+    policy_element = _policy.get_policy('/', policy_file_path)
+    context_element = policy_element.find('contexts/context')
+    context_element.attrib['path'] = identity
+
+    permissions_path = os.path.join(key_dir, 'permissions.xml')
+    _permission.create_permission_file(permissions_path, _utilities.domain_id(), policy_element)
+
+    signed_permissions_path = os.path.join(key_dir, 'permissions.p7s')
+    keystore_permissions_ca_key_path = os.path.join(
+        _keystore.get_keystore_private_dir(keystore_path), 'permissions_ca.key.pem')
+    _utilities.create_smime_signed_file(
+        keystore_ca_cert_path,
+        keystore_permissions_ca_key_path,
+        permissions_path,
+        signed_permissions_path
+    )
+
+    return True
+
+
+def list_keys(keystore_path):
+    contexts_path = _keystore.get_keystore_context_dir(keystore_path)
+    if not os.path.isdir(keystore_path):
+        raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), keystore_path)
+    if not os.path.isdir(contexts_path):
+        return True
+    for name in os.listdir(contexts_path):
+        if os.path.isdir(os.path.join(contexts_path, name)):
+            print(name)
+    return True
+
+
+def _is_key_name_valid(name):
```

Comment:
Given that `create_key` is public, maybe this is worth being public too?

Reply:
My understanding from https://github.com/ros2/sros2/pull/188#issuecomment-609054839 is that all API not currently used outside the module is made non-public.
After all the refactor PRs are merged we'll do another pass to decide which additional ones to make public

Reply:
Alright sounds good.